openpype/hosts/traypublisher/plugins/publish/validate_colorspace_look.py

```diff
@@ -0,0 +1,70 @@
+import pyblish.api
+
+from openpype.pipeline import (
+    publish,
+    PublishValidationError
+)
+
+
+class ValidateColorspaceLook(pyblish.api.InstancePlugin,
+                             publish.OpenPypePyblishPluginMixin):
+    """Validate colorspace look attributes"""
+
+    label = "Validate colorspace look attributes"
+    order = pyblish.api.ValidatorOrder
+    hosts = ["traypublisher"]
+    families = ["ociolook"]
+
+    def process(self, instance):
+        create_context = instance.context.data["create_context"]
+        created_instance = create_context.get_instance_by_id(
+            instance.data["instance_id"])
+        creator_defs = created_instance.creator_attribute_defs
+
+        ociolook_items = instance.data.get("ocioLookItems", [])
+
+        for ociolook_item in ociolook_items:
+            self.validate_colorspace_set_attrs(ociolook_item, creator_defs)
+
+    def validate_colorspace_set_attrs(self, ociolook_item, creator_defs):
+        """Validate colorspace look attributes"""
+
+        self.log.debug(f"Validate colorspace look attributes: {ociolook_item}")
+        self.log.debug(f"Creator defs: {creator_defs}")
+
+        check_keys = [
+            "working_colorspace",
+            "input_colorspace",
+            "output_colorspace",
+            "direction",
+            "interpolation"
+        ]
+        not_set_keys = []
+        for key in check_keys:
+            if ociolook_item[key]:
+                # key is set and it is correct
+                continue
+
+            def_label = next(
+                (d_.label for d_ in creator_defs if key == d_.key),
+                None
+            )
```

Comment:
If you would do this once outside the for loop: 
`creator_defs_by_key = {def.key: def.label for def in creator_defs}`

Then inside the for loop you can just do:
```python
def_label = creator_defs_by_key.get(key)
```
You can then also just log `def_attrs` below by just logging `creator_defs_by_key` instead.


Although I'm not entirely sure what you're trying to achieve here.

Reply:
done